engine/class_modules/priest/sc_priest_holy.cpp

```diff
@@ -12,130 +12,359 @@ namespace priestspace
+        if ( priest().talents.holy.apotheosis.enabled() && priest().buffs.apotheosis->up() )
+        {
+          chastise_cdr *= ( 1 + priest().talents.holy.apotheosis->effectN( 1 ).percent() );
+        }
+        if ( priest().talents.holy.light_of_the_naaru.enabled() )
+        {
+          chastise_cdr *= ( 1 + priest().talents.holy.light_of_the_naaru->effectN( 1 ).percent() );
+        }
+        sim->print_debug( "{} adjusted cooldown of Chastise, by {}, with harmonious_apparatus: {}, apotheosis: {}",
+                          priest(), chastise_cdr, priest().talents.holy.harmonious_apparatus.enabled(),
+                          ( priest().talents.holy.apotheosis.enabled() && priest().buffs.apotheosis->up() ) );
+
+        priest().cooldowns.holy_word_chastise->adjust( -chastise_cdr );
+      }
+      if ( child_searing_light && priest().buffs.divine_image->up() )
+      {
+        sim->print_debug( "searing_light triggered by holy_fire" );
+        for ( int i = 1; i <= priest().buffs.divine_image->stack(); i++ )
+        {
+          child_searing_light->execute();
+        }
+      }
+    }
 
-    priest().adjust_holy_word_serenity_cooldown();
+    if ( child_burning_vehemence )
+    {
+      child_burning_vehemence->trigger( s->result_amount );
+    }
+  }
+};
+
+struct holy_word_chastise_t final : public priest_spell_t
+{
+  propagate_const<searing_light_t*> child_searing_light;
+  holy_word_chastise_t( priest_t& p, util::string_view options_str )
+    : priest_spell_t( "holy_word_chastise", p, p.talents.holy.holy_word_chastise ), child_searing_light( nullptr )
+  {
+    parse_options( options_str );
+    if ( priest().talents.holy.divine_image.enabled() )
+    {
+      child_searing_light = new searing_light_t( priest() );
+    }
+  }
+  double cost() const override
+  {
+    if ( priest().buffs.apotheosis->check() )
+    {
+      return 0;
+    }
+
+    return priest_spell_t::cost();
+  }
+  double composite_da_multiplier( const action_state_t* s ) const override
+  {
+    double d = priest_spell_t::composite_da_multiplier( s );
+    if ( priest().buffs.divine_word->check() )
+    {
+      d *= 1.0 + priest().talents.holy.divine_word->effectN( 1 ).percent();
+      sim->print_debug( "divine_favor increasing holy_word_chastise damage by {}, total: {}",
+                        priest().talents.holy.divine_word->effectN( 1 ).percent(), d );
+    }
+    return d;
+  }
+
+  void execute() override
+  {
+    priest_spell_t::execute();
+    if ( priest().talents.holy.divine_word.enabled() && priest().buffs.divine_word->check() )
+    {
+      priest().buffs.divine_word->expire();
+      priest().buffs.divine_favor_chastise->trigger();
+    }
+    if ( priest().talents.holy.divine_image.enabled() )
+    {
+      priest().buffs.divine_image->trigger();
+      // Activating cast also immediately executes searing light
+      sim->print_debug( "searing_light triggered by holy_word_chastise" );
```

Comment:
im assuming this might be done for tracking purposes? not sure why you need this print statement though. Perhaps you could pass in the action name when triggering searing light and then do something with it to make the reporting clear?

Reply:
Added a proc to track divine_image instead, removed these debug statements.